### PR TITLE
feat(analysis): add period comparison, monthly trends, and cash position tools

### LINKED ISSUE #83: feat: Add period comparison and monthly trend analysis tools
## 🎯 Goal

Add server-side financial analysis tools that compute period comparisons, monthly trends, and cash position summaries — eliminating the need for LLM-side math.

**Definition of Done:**

- [ ] 3 new analysis tools compute all aggregations server-side
- [ ] LLM receives pre-calculated diffs, percentages, and summaries
- [ ] Common multi-call analysis patterns reduced to single tool calls

### Why This Matters

Common financial analysis patterns require multiple API calls and LLM-side computation:

1. **Year-over-Year comparison**: LLM must call `freee_get_profit_loss` twice, then calculate differences and percentages itself
2. **Monthly trends**: LLM must call `freee_get_profit_loss` 12 times (once per month) to build a trend
3. **Cash flow projection**: Requires combining walletable balances + unpaid invoices + unsettled deals

These patterns waste API calls and force computation into the LLM (which is unreliable for math).

### Current State

- LLM must make multiple sequential API calls for comparisons
- LLM computes percentage changes and trend analysis (error-prone)
- No way to get a consolidated cash position view

### Desired State

- Single tool call returns pre-computed period comparison with diffs
- Single tool call returns monthly trend data with summary statistics
- Single tool call returns consolidated cash position

### User Stories

- As an LLM user, I want to compare this year's P&L to last year's in one call so that I can identify key changes
- As an LLM user, I want monthly revenue trends so that I can spot patterns without 12 API calls
- As an LLM user, I want a cash position summary so that I can assess financial health quickly

### Functional Requirements

1. **`freee_compare_periods`** — Period comparison with automatic diff calculation
   ```typescript
   // Input
   { companyId, reportType: 'profit_loss' | 'balance_sheet',
     period1: { fiscalYear, startMonth, endMonth },
     period2: { fiscalYear, startMonth, endMonth },
     breakdownDisplayType?: 'partner' | 'item' | 'section' | 'tag' }

   // Output: Pre-calculated comparison
   { period1: { revenue: 1000000, ... },
     period2: { revenue: 1200000, ... },
     changes: { revenue: { amount: 200000, percentage: 20.0 }, ... },
     highlights: [
       { item: 'Revenue', change: '+20%', significance: 'high' },
       { item: 'SGA Expenses', change: '+5%', significance: 'low' }
     ] }
   ```
   Server-side: makes 2 API calls, computes all diffs and percentages.

2. **`freee_monthly_trends`** — Monthly P&L/BS trend in one call
   ```typescript
   // Input
   { companyId, fiscalYear, reportType: 'profit_loss' | 'balance_sheet',
     months?: number[] }  // defaults to all 12 months

   // Output: Monthly data array
   { months: [
       { month: 4, revenue: 100000, cogs: 40000, grossProfit: 60000, operatingProfit: 30000 },
       { month: 5, revenue: 120000, cogs: 45000, grossProfit: 75000, operatingProfit: 40000 },
       ...
     ],
     summary: { avgRevenue: 110000, maxRevenue: { month: 5, value: 120000 }, trend: 'increasing' } }
   ```
   Server-side: makes N API calls (one per month), extracts key metrics, computes summary statistics.

3. **`freee_cash_position`** — Current cash position overview
   ```typescript
   // Input
   { companyId }

   // Output
   { totalCash: 5000000,
     accounts: [{ name: '普通預金', balance: 4500000 }, { name: '現金', balance: 500000 }],
     receivables: { total: 2000000, overdue: 300000, count: 15 },
     payables: { total: 1500000, overdue: 100000, count: 8 },
     netPosition: 5500000 }
   ```
   Server-side: combines walletables + invoices(unpaid) + deals(unsettled).

### Technical Constraints

- **Technology**: TypeScript, Zod schemas, MCP SDK 1.x
- **Performance**: Monthly trends makes up to 12 API calls — consider parallel execution with `Promise.all()`
- **Math accuracy**: Use integer arithmetic (yen amounts) where possible to avoid floating point issues
- **Dependencies**: `freee_cash_position` requires walletables endpoint (#82)

### Functional

- [ ] `freee_compare_periods` registered with Zod schema
- [ ] `freee_monthly_trends` registered with Zod schema
- [ ] `freee_cash_position` registered with Zod schema
- [ ] All aggregation computed server-side
- [ ] Percentage changes and summary statistics calculated by program (not LLM)
- [ ] LLM receives structured, pre-analyzed data

### Testing

- [ ] Unit tests with mocked API responses for all 3 tools
- [ ] Edge cases: zero values, negative changes, missing months
- [ ] `npm run build && npm test` pass

### Affected Files

- `src/api/freeeClient.ts` — Add analysis methods (comparePeriods, monthlyTrends, cashPosition)
- `src/index.ts` — Register 3 new tools
- `src/schemas.ts` — Add Zod schemas
- `src/types/freee.ts` — Add TypeScript interfaces for analysis responses
- `src/__tests__/api/freeeClient.test.ts` — Unit tests with mocked responses

### Existing Patterns

- `freee_get_profit_loss` in FreeeClient shows how to call the P&L API
- Use existing `getInvoices()` and `getDeals()` for cash position data

### Technical Considerations

- Monthly trends: use `Promise.all()` for parallel API calls (12 months = 12 calls)
- Period comparison: extract key line items (revenue, COGS, gross profit, operating profit, net income) from P&L response structure
- Cash position: combine walletable balances with receivables/payables from invoices/deals
- Significance thresholds for highlights: >10% change = high, >5% = medium, else low

## 🚧 Risks & Mitigation

| Risk | Impact | Probability | Mitigation |
|------|--------|-------------|------------|
| 12 parallel API calls hitting rate limit | High | Medium | Sequential fallback, rate limit check before batch |
| P&L response structure varies by company settings | Medium | Medium | Defensive parsing, graceful handling of missing fields |
| Cash position accuracy (timing of data) | Medium | Low | Document that this is a snapshot, not real-time |

## 🔗 Related Work

- Depends on #82 (walletables endpoint) for `freee_cash_position`
- Depends on #80 (auto-pagination) for invoice/deal aggregation in cash position
- Related to #81 (cache) — master data cache improves performance of these tools

### 🟡 Recommended

- [ ] Test with Claude Desktop to verify tool descriptions guide LLM effectively
- [ ] Measure API call reduction in typical analysis sessions
- [ ] Compare period comparison accuracy against manual calculations
- [ ] Test monthly trends with companies that have non-April fiscal year starts

### ⚠️ Manual Verification

- [ ] Verify percentage calculations are correct
- [ ] Test cash position against known account balances
- [ ] Test with edge cases (new company with sparse data)

## 🏷️ Metadata

**Priority**: Medium
**Size**: L
**Type**: Feature
**Labels**: feature

---

🤖 AI-Optimized Issue
Generated with [Claude Code](https://claude.com/claude-code)

### PR DESCRIPTION
**Pull request by knishioka**
## Summary

Closes #83

- Add `freee_compare_periods` tool for comparing P&L or balance sheet between two periods with change amounts, percentages, and significance-ranked highlights
- Add `freee_monthly_trends` tool for fetching monthly financial data with automatic trend detection (increasing/decreasing/stable/fluctuating) and summary statistics
- Add `freee_cash_position` tool for aggregating cash balances, receivables (from invoices), payables (from deals), and net position

All computation is performed server-side to minimize LLM token usage. Parallel API calls (`Promise.all`) are used for multi-period fetches.

### Files changed

| File | Changes |
|------|---------|
| `src/schemas.ts` | Added `ComparePeriodsSchema`, `MonthlyTrendsSchema`, `CashPositionSchema` |
| `src/types/freee.ts` | Added type interfaces for all analysis response types |
| `src/api/freeeClient.ts` | Added `comparePeriods()`, `getMonthlyTrends()`, `getCashPosition()` methods + private helpers |
| `src/index.ts` | Registered 3 new MCP tools |
| `src/__tests__/api/analysis.test.ts` | 26 unit tests covering all methods and edge cases |
| `src/__tests__/index.test.ts` | Updated tool/schema counts from 24 to 27 |

## Test plan

- [x] 26 new unit tests for analysis methods (comparePeriods, getMonthlyTrends, getCashPosition)
- [x] All 285 tests pass (including existing tests)
- [x] TypeScript typecheck passes
- [x] ESLint passes (0 errors)
- [x] Pre-commit hooks pass (eslint, typecheck, gitleaks)
- [ ] Manual verification with freee sandbox account

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/api/freeeClient.ts [hunk 1]
```diff
@@ -1020,5 +1028,281 @@ export class FreeeClient {
...
+
+    const thirdSize = Math.max(1, Math.floor(values.length / 3));
+    const firstThird = values.slice(0, thirdSize);
+    const lastThird = values.slice(-thirdSize);
+
+    const firstAvg = firstThird.reduce((a, b) => a + b, 0) / firstThird.length;
+    const lastAvg = lastThird.reduce((a, b) => a + b, 0) / lastThird.length;
+
+    if (firstAvg === 0 && lastAvg === 0) return 'stable';
+    if (firstAvg === 0) return 'increasing';
```
[the remark this answers is not in the document]
Comment (knishioka) [reply src/api/freeeClient.ts]: Fixed in commit 576c879. computeTrend now correctly returns 'decreasing' when values go from zero to negative. Added a test case for this edge case.
